Code/autopkglib/github/__init__.py

```diff
@@ -89,6 +97,69 @@ def setup_token(self):
 
         self.token = token
 
+    def prepare_curl_cmd(self, method, accept, headers, data, temp_content):
+        """Assemble curl command and return it."""
+        curl_cmd = [
+            super(GitHubSession, self).curl_binary(),
+            "--location",
+            "--silent",
+            "--show-error",
+            "--fail",
+            "--dump-header",
+            "-",
+        ]
+
+        curl_cmd.extend(["-X", method])
+        curl_cmd.extend(["--header", "%s: %s" % ("User-Agent", "AutoPkg")])
+        curl_cmd.extend(["--header", "%s: %s" % ("Accept", accept)])
+
+        # Pass the GitHub token as a header
+        if self.token:
+            curl_cmd.extend(
+                ["--header", "%s: %s" % ("Authorization", "token %s" % self.token)]
+            )
+
+        super(GitHubSession, self).add_curl_common_opts(curl_cmd)
+
+        # Additional headers if defined
```

Comment:
This is already handled by `add_curl_common_opts`.

Reply:
No it's not. Look at the origin of the `headers` variable. 

`GitHubSession` is not a Processor but kind of an ad-hoc module. `add_curl_common_opts` method in superclass only adds headers from `self.env.get("request_headers")` object variable. `request_headers` is configured in Processors `input_variables`. `GitHubSession` does not have `input_variables` like Processor.

Currently calling `add_curl_common_opts()` is useful only for `curl_opts` set in `__init__()` constructor.